### PR TITLE
Component switch

### PR DESCRIPTION
**Pull request by checkmate-bitch**
created files for each component and implemented switching code on click

### CODE REVIEW

File: app/index.jsx [hunk 1]
```diff
@@ -17,10 +32,11 @@ let App = React.createClass({
     this.setState({ selectedElementIndex: clickedLiIndex })
   },
   render: function () {
-    let featureTitle = constants.elements[this.state.selectedElementIndex]
+    const featureTitle = constants.elements[this.state.selectedElementIndex]
+    const Component = componentCode[this.state.selectedElementIndex]
```
Comment (tamj0rd2) [inline app/index.jsx]: rename to `var BootstrapElement` or leave it as `BootstrapPre`?
Seems like a better name than just `Component`.

File: app/index.jsx [hunk 2]
```diff
@@ -2,8 +2,23 @@ var React = require('react')
 var ReactDOM = require('react-dom')
 const constants = require('./scripts/constants.js')
 
-var Nav = require('./components/nav.jsx')
-var BootstrapPre = require('./components/bootstrap-pre.jsx')
+var Navigation = require('./components/navigation.jsx')
```
Comment (tamj0rd2) [inline app/index.jsx:5]: Why did you rename this to Navigation?

File: app/index.jsx [hunk 3]
```diff
@@ -2,8 +2,23 @@ var React = require('react')
...
+const componentCode = [
```
Comment (tamj0rd2) [inline app/index.jsx:6]: Don't require the code into an array.

Couple things to do here:

Create a folder called app/components/elements
Move the elements you've created into the above folder (alerts-pre, button-groups-pre) etc
Create a file called app/components/elements.jsx
In elements.jsx, do something like:

```
module.exports = {
  Alert: require('./components/elements/alerts-pre.jsx),
  ButtonDropdown: require('./components/elements/button-dropdowns-pre.jsx),
  etc...
}

Then in index.jsx, you can do `var elements = require('./elements.jsx')`
From then on, you can then refer to each element by key. And if you really need to at any point, loop through the elements by key and value.

File: app/components/alerts-pre.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+var React = require("React");
+
+let Alerts = React.createClass({
```
Comment (tamj0rd2) [inline app/components/alerts-pre.jsx]: Remove the `s` from each of the class names. The class will only give you 1 Alert element, not multiple.

**Review by tamj0rd2 — CHANGES_REQUESTED** [review]
Make sure you install Eslint and EditorConfig. The code style we're using doesn't include semi-colons. Both of those plugins will catch most of the code style mistakes in your code.

The folder structure just needs a little rethinking, and I'm not sure why you renamed certain things. I go into more depth below though. Other than that, good work.

File: app/components/alerts-pre.jsx (hunk 4 above)
Comment (checkmate-bitch) [reply app/components/alerts-pre.jsx]: Will correct that

File: app/index.jsx (hunk 1 above)
Comment (checkmate-bitch) [reply app/index.jsx]: That will be holding any one of the component code at a time. So I went with generic name. Should I change it?

File: app/index.jsx (hunk 3 above)
Comment (checkmate-bitch) [reply app/index.jsx:6]: Seems like too much work for something that can be done easily. I'll look into it, though

File: app/index.jsx (hunk 2 above)
Comment (checkmate-bitch) [reply app/index.jsx:5]: Because we have Nav, Navbar components. All the components are a part of the navigation in the view. So, I thought it would be apt

Comment (tamj0rd2) [reply app/index.jsx:5]: Oh fair enough, I didn't spot that. Good idea

File: app/index.jsx (hunk 3 above)
Comment (tamj0rd2) [reply app/index.jsx:6]: It's not really. All you'll have to do is move all of the files you created into an elements folder
Then create an elements file in the components folder.
The copy paste the require statements into the elements file.

I can do it if you want, it'll only take a couple of minutes.

File: app/index.jsx (hunk 1 above)
Comment (tamj0rd2) [reply app/index.jsx]: I think it's a good idea to change it, just to make it more obvious what it's supposed to be. It'll be more readable that way.

`Component` could mean anything, since react classes are also known as components.
